pymatgen/electronic_structure/boltztrap2.py

```diff
@@ -56,16 +56,162 @@
+            self.cbm_idx = min(bs_obj.get_cbm()['band_index'][Spin.up] +
+                               bs_obj.get_cbm()['band_index'][Spin.down])
+            self.vbm = bs_obj.get_vbm()['energy']
+            self.cbm = bs_obj.get_cbm()['energy']
+        else:
+            self.vbm_idx = None
+            self.cbm_idx = None
+            self.vbm = self.felrmi
+            self.cbm = self.fermi
+
+        if nelect:
+            self.nelect_all = nelect
+        elif self.vbm_idx:
+            self.nelect_all = self.vbm_idx + self.cbm_idx + 1
+        else:
+            raise BoltztrapError('nelect must be given.')
+
+    @classmethod
+    def from_file(cls, vasprun_file):
+        """Get a vasprun.xml file and return a PMGLoader"""
+        vrun_obj = Vasprun(vasprun_file, parse_projected_eigen=True)
+        return cls(vrun_obj)
+
+    def get_lattvec(self):
+        """
+        :return: The lattice vectors.
+        """
+        try:
+            self.lattvec
+        except AttributeError:
+            self.lattvec = self.atoms.get_cell().T * units.Angstrom
+        return self.lattvec
+
+    def get_volume(self):
+        """
+        :return: Volume
+        """
+        try:
+            self.UCvol
+        except AttributeError:
+            lattvec = self.get_lattvec()
+            self.UCvol = np.abs(np.linalg.det(lattvec))
+        return self.UCvol
+
+    def bandana(self, emin=-np.inf, emax=np.inf):
+        """Cut out bands outside the range (emin,emax)"""
+        bandmin = np.min(self.ebands_all, axis=1)
+        bandmax = np.max(self.ebands_all, axis=1)
+        ntoolow = np.count_nonzero(bandmax <= emin)
+        accepted = np.logical_and(bandmin < emax, bandmax > emin)
+        # self.data_bkp = np.copy(self.data.ebands)
+        self.ebands = self.ebands_all[accepted]
+
+        self.proj = {}
+        if self.proj_all:
+            if len(self.proj_all) == 2:
+                h = int(len(accepted) / 2)
+                self.proj[Spin.up] = self.proj_all[
+                    Spin.up][:, accepted[:h], :, :]
+                self.proj[Spin.down] = self.proj_all[
+                    Spin.down][:, accepted[h:], :, :]
+            elif len(self.proj_all) == 1:
+                self.proj[Spin.up] = self.proj_all[Spin.up][:, accepted, :, :]
+
+        if self.mommat_all:
+            self.mommat = self.mommat[:, accepted, :]
+        # Removing bands may change the number of valence electrons
+        if self.nelect_all:
+            self.nelect = self.nelect_all - self.dosweight * ntoolow
+
+        return accepted
+
+
+class BandstructureLoader:
+    """Loader for Bandstructure object"""
+
+    def __init__(self,
+                 bs_obj,
+                 structure=None,
+                 nelect=None,
+                 mommat=None,
+                 magmom=None):
         """
         :param bs_obj:
         :param structure:
         :param nelect:
         :param spin:
+        :param momat:
```

Comment:
I need proper documentation for every parameter.

Reply:
right, done!